shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -70,14 +72,15 @@ @interface FlutterPlatformPlugin ()
  *        info.plist makes this value to be false.
  */
 @property(nonatomic, assign) BOOL enableViewControllerBasedStatusBarAppearance;
+@property(nonatomic, weak) FlutterEngine* engine;
```

Comment:
Can we also fix up `-initWithEngine` to take a normal `FlutterEngine` instead of a `fml::WeakNSObject`?

https://github.com/flutter/engine/blob/2fce288b515745f204897c1714bfd1fc37b61d29/shell/platform/darwin/ios/framework/Source/FlutterEngine.mm#L676
```objc
  _platformPlugin.reset([[FlutterPlatformPlugin alloc] initWithEngine:self]);
```

Reply:
Yes -- I should have tagged that with the same TODO as the other two. Once all translation units importing the header are ARC-ified, I'll remove it.

Reply:
[The magic of C++ implicit conversions!](https://github.com/flutter/engine/blob/af39fda507474d78699f760ad5f81f43dcb846f4/fml/memory/weak_ptr.h#L92-L102)

Reply:
> [The magic of C++ implicit conversions!](https://github.com/flutter/engine/blob/af39fda507474d78699f760ad5f81f43dcb846f4/fml/memory/weak_ptr.h#L92-L102)

I think this guy: 
https://chromium.googlesource.com/chromium/src.git/+/refs/tags/94.0.4606.43/base/ios/weak_nsobject.h#175

Reply:
Oh sorry yeah didn't look at the class before I looked it up! You got the right class, but it's this line:
https://chromium.googlesource.com/chromium/src.git/+/refs/tags/94.0.4606.43/base/ios/weak_nsobject.h#179

We're converting _from_ a WeakNSObject<T> _to_ a T.

Reply:
TL;DR this is all horrible and getting rid of it is all a huge win.